### PR TITLE
[CELEBORN-864][DOC] Document on blacklist

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
As title.

### Why are the changes needed?
As title.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @pan3793 @cfmcgrady @FMX

File: docs/developers/blacklist.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+excludes a `Worker` in the following scenarios:
+
+- Receives Revive request and the cause is critical
+- Fail to send RPC to a `Worker`
+- From `Master`'s excluded list, carried in the heartbeat response
+
+`LifecycleManager` will remove `Worker` from the excluded list in the following scenarios:
+
+- For critical causes, when timeout expires (defaults to 180s)
+- for non-critical causes, when it's not in `Master`'s blacklist
```
Comment (zhongqiangczq) [inline docs/developers/blacklist.md:81]:
```suggestion
- For non-critical causes, when it's not in `Master`'s blacklist
```

File: mkdocs.yml [hunk 2]
```diff
@@ -93,5 +93,4 @@ nav:
         - LifecycleManager: developers/lifecyclemanager.md
         - ShuffleClient: developers/shuffleclient.md
       - Fault Tolerant: developers/faulttolerant.md
-#      - ReadData: developers/readdata.md
-#      - Slots Allocation: developers/slotsallocation.md
+      - Blacklist: developers/blacklist.md
```
Comment (zhongqiangczq) [inline mkdocs.yml:96]: can still use blacklist concept as we already rename blacklist to excluded

Comment (pan3793) [reply mkdocs.yml:96]: should be "worker exclusion"

**Comment by waitinfuture** [comment]
ping @zhongqiangczq @pan3793 @FMX comments addressed, please take a look again

**Comment by waitinfuture** [comment]
Thanks, merging to main.

**Pull request closed without merging**